### PR TITLE
adding helm implementation

### PR DESCRIPTION
**Pull request by vanajamukkara**
Adding HelmParams class
OperatorParams, TraefikParams extend HelmParams
Implementation added in Helm class for install, upgrade, delete methods.
Operator, Traefik extend Helm now.
ItSimpleOperatorValidation - installs operator and verifies operator pod is running

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 1]
```diff
@@ -65,69 +64,89 @@ public void testInstallingOperator() {
...
+    logger.info(String.format("Got a new namespace called %s", opNamespace));
+
+    final String domainNamespace1 = assertDoesNotThrow(TestActions::createUniqueNamespace,
+        "Failed to create unique namespace due to ApiException");
+    logger.info(String.format("Got a new namespace called %s", domainNamespace1));
+
+    final String domainNamespace2 = assertDoesNotThrow(TestActions::createUniqueNamespace,
         "Failed to create unique namespace due to ApiException");
-    logger.info(String.format("Got a new namespace called %s", namespace));
+    logger.info(String.format("Got a new namespace called %s", domainNamespace2));
```
Comment (ddsharpe) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:78]: -> "Created a new namespace called ..."
Doesn't the logger facade in the operator support String formats like "blah blah {0}"?  Shouldn't we use the logging facade in operator rather than java.util.logging directly?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 2]
```diff
@@ -65,69 +64,89 @@ public void testInstallingOperator() {
...
 
-    // Create a service account for the unique namespace
-    final String serviceAccountName = namespace + "-sa";
-    assertDoesNotThrow(
-        () -> TestActions.createServiceAccount(new V1ServiceAccount()
-            .metadata(new V1ObjectMeta().namespace(namespace).name(serviceAccountName))));
+    // Create a service account for the unique opNamespace
+    final String serviceAccountName = opNamespace + "-sa";
+    assertDoesNotThrow(() -> TestActions.createServiceAccount(new V1ServiceAccount()
+        .metadata(new V1ObjectMeta().namespace(opNamespace).name(serviceAccountName))));
```
Comment (ddsharpe) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:83]: Every assertion should have a short message that would help someone debug the issue should it occur.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 3]
```diff
@@ -5,86 +5,175 @@
...
+        .as("make sure namespace is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getReleaseName())
+        .as("make sure releaseName is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getRepoUrl() != null && params.getChartName() == null)
```
Comment (ddsharpe) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:35]: These combinations are easier to read and debug if they are separate. Like:
assertNotNull(params.getRepoUrl(), "repo URL was not provided") 
or
assertThat(params.getRepoUrl())
.as("repo URL was not provided")
. isNotNull()
.isNotEmpty();

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 4]
```diff
@@ -5,86 +5,175 @@
...
+        .as("make sure chart name is not empty or null when repo url is provided")
+        .isFalse();
+
+    assertThat(params.getRepoUrl() == null && params.getChartDir() == null)
```
Comment (ddsharpe) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:39]: same as previous (combo check)

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 5]
```diff
@@ -5,86 +5,175 @@
...
+        .as("make sure repo url, chart name and chart dir are not empty or null. "
+            + "repo url, chart name or chart dir must be provided")
+        .isFalse();
+
+    //chart reference to be used in helm install
+    String chartRef = params.getChartDir();
+
+    // use repo url as chart reference if provided
+    if (params.getChartName() != null && params.getRepoUrl() != null) {
```
Comment (ddsharpe) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:48]: combo check

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 6]
```diff
@@ -5,86 +5,175 @@
...
+        .isNotEmpty();
+
+    //chart reference to be used in helm upgrade
+    String chartDir = params.getChartDir();
+
+    logger.info(String.format("Upgrade application in namespace %s using chart %s",
+        namespace, chartDir));
+
+    // build helm upgrade command
+    StringBuffer upgradeCmd = new StringBuffer("helm upgrade ")
```
Comment (ddsharpe) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:101]: Its not a big deal, but this seems harder to read than necessary.  Why not use String.format?
String upgradeCmd = "helm upgrade {0} {1} --namespace {3}";

appendValues( params.getValues(),
   String.format(upgradeCmd, params.getReleaseName(), chartDir, params.getNamespace());

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 1 above)
Comment (markxnelson) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:78]: this is the logger injected by `LoggedTest` isn't it?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 7]
```diff
@@ -65,69 +64,89 @@ public void testInstallingOperator() {
...
+                  condition.getElapsedTimeInMS(),
+                  condition.getRemainingTimeInMS())))
+          // and here we can set the maximum time we are prepared to wait
+          .await().atMost(5, MINUTES)
+          // operatorIsRunning() is one of our custom, reusable assertions
+          .until(operatorIsRunning(opNamespace));
+    } finally {
+
+      // uninstall operator release
+      assertThat(TestActions.uninstallOperator(opParams))
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:124]: @rjeberhard @ddsharpe interested in your opinion on this --- seems like assertions in the finally is somehow wrong -- if these are important, should they not be part of the test itself?   not sure about assertions in a finally block.....

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java [hunk 8]
```diff
@@ -3,69 +3,10 @@
...
-                        .install();
-    }
-    return success;
-  }
-
-  public static boolean upgrade(OperatorParams params) {
-    return true;
-  }
 
+public class Operator extends Helm {
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java:8]: i need to take a much closer look at this, my gut reaction is that extension here is not the best solution

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/OperatorParams.java [hunk 9]
```diff
@@ -7,9 +7,11 @@
 import java.util.List;
 import java.util.Objects;
 
+import oracle.weblogic.kubernetes.actions.impl.primitive.HelmParams;
+
 // All parameters needed to install Operator from test
 
-public class OperatorParams {
+public class OperatorParams extends HelmParams {
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/OperatorParams.java:14]: same here

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java (hunk 8 above)
Comment (markxnelson) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java:8]: seems like an operator is much more than just a set of helm params
maybe we could `implement HelmRelease` 
or have a variable that is a `HelmParams` 

i think an action impl should *never* extend a primitive
they should *use* primitives, not "be" them... imo

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/TraefikParams.java [hunk 10]
```diff
@@ -5,13 +5,13 @@
 
 import java.util.HashMap;
 
+import oracle.weblogic.kubernetes.actions.impl.primitive.HelmParams;
+
 // All parameters needed to install Traefik Operator
 
-public class TraefikParams {
+public class TraefikParams extends HelmParams {
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/TraefikParams.java:12]: same comment about inheritance

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 11]
```diff
@@ -5,86 +5,175 @@
...
-    this.values = values;
-    return this;
-  }
+    // add all the parameters
+    appendValues(params.getValues(), installCmd);
 
-  public String getChartName() {
-    return chartName;
-  }
+    installCmd.append(" && helm list");
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:64]: i think this is not ideal - if it fails, how will we know which of the two commands failed?
i think it would be better to run each one separately and then inspect the results
i know is it "more work" - but i think it is much clearer and will produce more reliable code to do if we do it that way

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 12]
```diff
@@ -5,86 +5,175 @@
...
+    return exec(installCmd.toString());
 
-  public String getReleaseName() {
-    return releaseName;
   }
 
-  public String getNamespace() {
-    return namespace;
+  /**
+   * upgrade a helm release
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:71]: captials and periods please

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 13]
```diff
@@ -5,86 +5,175 @@
...
+   * @param params the parameters to helm as values
+   * @return true on success, false otherwise
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:73]: nitpick: if you say it returns true on success, no need to say false otherwise, that is pretty clear :)

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 14]
```diff
@@ -5,86 +5,175 @@
...
+   */
+  public static boolean upgrade(HelmParams params) {
+    String namespace = params.getNamespace();
+
+    // assertions for required parameters
+    assertThat(namespace)
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:79]: these are really nice

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 6 above)
Comment (markxnelson) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:101]: yeah i think `String.format()` would be a lot cleaner to read

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 15]
```diff
@@ -5,86 +5,175 @@
...
+
+    // build helm upgrade command
+    StringBuffer upgradeCmd = new StringBuffer("helm upgrade ")
+        .append(params.getReleaseName()).append(" ").append(chartDir)
+        .append(" --namespace ").append(params.getNamespace());
+
+    // add all the parameters
+    appendValues(params.getValues(), upgradeCmd);
+
+    upgradeCmd.append(" && helm list");
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:108]: same comment as before

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 16]
```diff
@@ -5,86 +5,175 @@
...
+    // run the command
+    return exec(upgradeCmd.toString());
   }
 
-  public String getRepoUrl() {
-    return repoUrl;
+  /**
+   * uninstall a helm release
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:114]: capitals, period.
Helm has a capital H

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 17]
```diff
@@ -5,86 +5,175 @@
...
+  public static boolean addRepo(String chartName, String repoUrl) {
+    String addRepoCmd = "helm add repo " + chartName + " " + repoUrl;
+    return exec(addRepoCmd);
   }
 
-  public boolean install() {
-    StringBuffer installCmd = new StringBuffer("helm install ")
-                              .append(releaseName).append(" ").append(chartName);
+  /**
+   * append the values to the given string buffer
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:146]: cap, period

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 18]
```diff
@@ -5,86 +5,175 @@
...
+            .append("=")
+            .append(value.toString().replaceAll("\\[", "{").replaceAll("\\]", "}").replace(" ",""))
+            .append("\""));
   }
 
-  public boolean addRepo() {
-    String addRepoCmd = "helm add repo " + chartName + " " + repoUrl;
-    // execute addRepoCmd
+  /**
+   * Executes the given command
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:160]: period

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 19]
```diff
@@ -5,86 +5,175 @@
...
+   * @param command the command to run
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:161]: nitpick: if this method "executes" a command, then it is probably nicer to say "the command to execute"

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java [hunk 20]
```diff
@@ -5,86 +5,175 @@
...
+   * @return true on success, false otherwise
+   */
+  private static boolean exec(String command) {
+    logger.info("Running command - \n" + command);
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:165]: These feel like the should maybe be debug level messages, not info
@rjeberhard @ddsharpe what do you think?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/HelmParams.java [hunk 21]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.actions.impl.primitive;
+
+import java.util.HashMap;
+
+// All common parameters needed to install a helm application
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/HelmParams.java:8]: capital H

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/HelmParams.java [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+
+public abstract class HelmParams {
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/HelmParams.java:10]: why an abstract class?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 1 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:78]: @ddsharpe will change the message. For logger, it should be fixed in LoggedTest as Mark said.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 2 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:83]: will fix the assertions to have a message. Thanks @markxnelson for the assertions test in the other PR.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 3 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:35]: @ddsharpe here I need to check chartName is also provided only when repoUrl is provided. Otherwise chartName is not required. 

When repoUrl is provided, here its adding a chart repository, for that both repoUrl  and chartName are required. hence the combination.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 4 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:39]: helm install command uses repo url or chart dir, at least one of these must be provided. Both are not required, hence the combination.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 6 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:101]: will modify to use String.format()

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 23]
```diff
@@ -65,69 +64,89 @@ public void testInstallingOperator() {
...
+            .serviceAccount(serviceAccountName);
 
-    //ToDO: use Junit 5 assertions
     try {
-      installOperator(opParams);
-    } catch (ApiException e) {
-      e.printStackTrace();
-      fail("Failed to install Operator due to exception" + e.getMessage());
-    }
+      assertThat(installOperator(opParams)).as("Operator installation failed").isTrue();
```
[the remark this answers is not in the document]
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:95]: will fix the assertions

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 24]
```diff
@@ -65,69 +64,89 @@ public void testInstallingOperator() {
...
+    } finally {
+
+      // uninstall operator release
+      assertThat(TestActions.uninstallOperator(opParams))
+          .as("Operator uninstall failed")
+          .isTrue();
+
+      // Delete service account from unique opNamespace
+      assertDoesNotThrow(
+          () -> TestActions.deleteServiceAccount(new V1ServiceAccount()
```
[the remark this answers is not in the document]
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:130]: will do

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 11 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:64]: will do

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 3 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:35]: checking for more readable ways ...

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 25]
```diff
@@ -65,69 +64,89 @@ public void testInstallingOperator() {
...
 
     OperatorParams opParams =
         new OperatorParams().releaseName("weblogic-operator")
-            .namespace(namespace)
-            .image("weblogic-kubernetes-operator:test_itsimpleoperator")
-            .domainNamespaces(Arrays.asList("domainns1", "domainns2"))
-            .serviceAccount("opns1-sa");
+            .namespace(opNamespace)
+            .chartDir("../kubernetes/charts/weblogic-operator")
+            .image("phx.ocir.io/weblogick8s/weblogic-kubernetes-operator:develop")
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:90]: This is the part I was discussing in the channel... Depending on the sort of cluster, the image may need to be pushed or not.  But, I guess this can wait and need not be part of this pull request.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 7 above)
Comment (rjeberhard) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:124]: Hmmm... I agree that that does seem incorrect.  If the point is to test uninstallation then this should be a test; otherwise, I'd have clean-up in an extension or "after" lifecycle method.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 26]
```diff
@@ -64,24 +64,24 @@ public static boolean scaleDomain(String domainUID, String clusterName, int numO
...
   }
 
   // ----------------------   domain  -----------------------------------
 
   /**
    * Create domain custom resource from the given domain yaml file.
    *
-   * @param domainUID - unique domain identifier
-   * @param namespace - name of namespace
+   * @param domainUID  - unique domain identifier
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java:83]: You don't need the formatting here (the hyphen)... that will generated Javadoc that starts with this symbol.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java (hunk 8 above)
Comment (rjeberhard) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java:8]: Agree.  HelmParams or the other choice seems more appropriate.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 1 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java]: changed the message

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 2 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 6 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 7 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java]: moved to after lifecycle method.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java (hunk 8 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Operator.java]: Agree. Removed inheritance and modified impl classes to call Helm instead.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/OperatorParams.java (hunk 9 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/OperatorParams.java]: Removed inheritance

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 23 above)
[the remark this answers is not in the document]
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java]: fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 24 above)
[the remark this answers is not in the document]
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 3 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java]: @markxnelson @rjeberhard any suggestions please ^^ ?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 15 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java]: modified

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/HelmParams.java (hunk 22 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/HelmParams.java]: Now as I removed inheritance, don't need the abstract method in this class anymore and this class is not abstract now.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java (hunk 26 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java]: removed the formatting

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 27]
```diff
@@ -116,18 +148,70 @@ public void testInstallingOperator() {
...
+
+  private String createUniqueNamespaceInTryBlock() {
+    String namespace = null;
+    try {
+      namespace = createUniqueNamespace();
+    } catch (Exception e) {
+      assertThat(e)
+          .as("Test that createUniqueNamespace does not throw an exception")
+          .withFailMessage("createUniqueNamespace() threw an unexpected exception")
+          .isNotInstanceOf(Exception.class);
```
Comment (markxnelson) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:212]: what class should it be?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java [hunk 28]
```diff
@@ -116,18 +148,70 @@ public void testInstallingOperator() {
...
+          () -> deleteNamespace(opNamespace))
+          .as("Test that deleteNamespace doesn not throw an exception")
+          .withFailMessage("deleteNamespace() threw an exception")
+          .doesNotThrowAnyException();
+      logger.info("Deleted namespace: " + opNamespace);
+    }
+
+  }
+
+  private String createUniqueNamespaceInTryBlock() {
```
Comment (rjeberhard) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java:204]: Why the strange name?  It's because you don't want the same name as TestActions.createUninqueNamespace(), correct :(?  The "InTryBlock" portion of the name looks strange.  Can this just be createNamespace()?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java (hunk 20 above)
Comment (rjeberhard) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java:183]: Let's get some tests running and then prune these back.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java (hunk 27 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleOperatorValidation.java]: Just saw Ryan's comment. We can use PR#1537 as this PR is closed, we have the same code.
